src/class/video/video_device.c

```diff
@@ -759,10 +759,15 @@ static bool _open_vs_itf(uint8_t rhport, videod_streaming_interface_t *stm, uint
   for (i = 0; i < TU_ARRAY_SIZE(stm->desc.ep); ++i) {
     uint_fast16_t ofs_ep = stm->desc.ep[i];
     if (!ofs_ep) break;
-    uint8_t  ep_adr = _desc_ep_addr(desc + ofs_ep);
-    usbd_edpt_close(rhport, ep_adr);
-    stm->desc.ep[i] = 0;
-    TU_LOG_DRV("    close EP%02x\r\n", ep_adr);
+    tusb_desc_endpoint_t const *ep = (tusb_desc_endpoint_t const*)(desc + ofs_ep);
+    /* Only ISO endpoints needs to be closed */
+    if(ep->bmAttributes.xfer == TUSB_XFER_ISOCHRONOUS) {
+#ifndef TUP_DCD_EDPT_ISO_ALLOC
```

Comment:
Maybe this `#ifndef` can cover this whole for loop, since bulk endpoint doesn;t change.

Reply:
I agree with your comment. I believe that the `#ifndef` can be moved to outside the `for` statement.

Reply:
As described below, `ep[1]` is limited to bulk for still image capture, so there is no need for a `for` statement.